olive/cli/perf_tuning.py

```diff
@@ -0,0 +1,312 @@
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+# --------------------------------------------------------------------------
+import json
+import logging
+import tempfile
+from argparse import ArgumentParser
+from copy import deepcopy
+from pathlib import Path
+from typing import Dict
+
+import yaml
+
+from olive.auto_optimizer.template_mapping import PERF_TUNING_TEMPLATE
+from olive.cli.base import BaseOliveCLICommand
+from olive.common.utils import set_nested_dict_value, set_tempdir
+from olive.workflows import run as olive_run
+
+logger = logging.getLogger(__name__)
+
+
+class PerfTuningCommand(BaseOliveCLICommand):
+
+    @staticmethod
+    def register_subcommand(parser: ArgumentParser):
+        sub_parser = parser.add_parser(
+            "tune-session-params",
+            help=(
+                "Automatically tune the session parameters for a given onnx model. "
+                "Currently, for onnx model converted from huggingface model and used for "
+                "generative tasks, user can simply provide the --model onnx_model_path "
+                "--hf_model_name hf_model_name --device device_type to get the tuned session parameters."
+            ),
+        )
+        # model options
+        model_group = sub_parser.add_argument_group("model options")
+        model_group.add_argument(
+            "--model",
+            required=True,
+            help="Onnx input model path.",
+        )
+
+        # dataset options
+        dataset_group = sub_parser.add_argument_group("dataset options")
+        dataset_group.add_argument(
+            "--data_config_path",
+            type=str,
+            help="Path to the data config file. It allows to customize the data config for the model.",
+        )
+        dataset_group.add_argument(
+            "--predict_with_kv_cache",
+            action="store_true",
+            help="Whether to use key-value cache for perf_tuning",
+        )
+        dataset_group.add_argument(
```

Comment:
maybe the hf dataset params should be in a separate group than the `dataset options` group? If data config path is provided, all of the other options are not relevant anymore. Don't know if it's possible to create two mutually exclusive groups but atleast grouping them separately would be less confusing.

Reply:
Updated.